enumerable-method.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+module Enumerable
+  # my_each
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
+    end
+    # Analysis:
+    # Self represent the current object, returns the elements of the array
+    # Yield asks for a block
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    (0...length).each do |i|
+      yield self[i], i
+    end
+  end
+
+  # my_select
+  def my_select
+    # new array
+    select_arr = []
+
+    my_each do |i|
+      select_arr << i if yield(i)
+    end
+    select_arr
+    # select: Returns a new array containing all elements of enum for which the given block returns a true value.
+  end
+
+  # my_all?
+  def my_all?
+    # all: The method returns true if ALL elements are true (or empty array).
+
+    length.times do |i|
+      return false unless yield self[i]
+    end
+    true
+  end
+
+  # my_any?
+  def my_any?
+    # any: The method returns true if AT LEAST one element is true (or non empty array).
+
+    selected = my_select do |i|
+      yield(i)
+    end
+    !selected.empty?
+  end
+
+  # my_none?
+  def my_none?
+    # none: The method returns true if NO elements are true (or empty array).
+
+    selected = my_select { |i| yield(i) }
+    selected.length <= 0
+  end
+
+  # my_count
+  def my_count
+    # count: The count method takes an enumerable collection and counts how many elements match the given criteria.
+    count = 0
+    my_each do |element|
+      count += 1 if !block_given? || yield(element)
+    end
+    count
+  end
+
+  # my_map
+  def my_map(proc = nil)
+    result = []
+    my_each do |element|
+      result << (block_given? ? yield(element) : proc.call(element))
+    end
+    result
+  end
+
+  # my_inject
+  def my_inject(initial = nil)
+    result = initial
+
+    my_each do |element|
+      result = if result.nil?
+                 element
+               else
+                 yield(result, element)
+               end
+    end
+    result
+  end
+end
```

Comment:
This behavior is expected  (when evaluating a range):

```
irb(main):026:0> (1..3).inject { |acc, n| acc + n }
=> 6
irb(main):027:0> (1..3).inject(1) { |acc, n| acc + n }
=> 7
irb(main):028:0> (1..3).inject(5) { |acc, n| acc + n }
=> 11
```

it injects the operation across the range  1+2+3 , starting to count from the injected number

----------------------

if we input a symbol as an argument, that operation should also be injected to our range (or array)

```
irb(main):029:0> (1..3).inject(:+)
=> 6
irb(main):030:0> [1,2].inject(:+)
=> 3



```